protocol/x/govplus/keeper/msg_server.go

```diff
@@ -15,3 +21,40 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (k msgServer) SlashValidator(
+	goCtx context.Context,
+	msg *types.MsgSlashValidator,
+) (*types.MsgSlashValidatorResponse, error) {
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
+	consAddr, err := sdk.ConsAddressFromBech32(msg.ValidatorAddress)
+	if err != nil {
+		return nil, types.ErrValidatorAddress
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	_, err = k.stakingKeeper.Slash(
+		ctx,
+		consAddr,
+		int64(msg.InfractionHeight),
```

Comment:
should the proto itself just use `int64/sint64`?

Reply:
We use uint32 elsewhere for height.